domain/application/state/types.go

```diff
@@ -196,6 +196,13 @@ type ipAddress struct {
 	DeviceID     string `db:"device_uuid"`
 }
 
+type spaceAddress struct {
+	Value     string         `db:"address_value"`
+	TypeID    int            `db:"type_id"`
+	ScopeID   int            `db:"scope_id"`
```

Comment:
Considering that we want the string version of these, I'd join against the table to get the string value. Then you don't need to unmarshal the value in code. Get the query to do the heavy lifting.